lms/djangoapps/courseware/views.py

```diff
@@ -367,6 +367,22 @@ def _index_bulk_op(request, course_key, chapter, section, position):
             user.id, unicode(course.id))
         return redirect(reverse('dashboard'))
 
+    # see if course has entrance exam enabled then it should be passed by user
+    # before accessing any other chapter/section of course
+    # Note that if the entrance exam feature flag has been turned off (default)
+    # then this check will always pass
+    if chapter and getattr(course, 'entrance_exam_enabled', False):
+        chapter_descriptor = course.get_child_by(lambda m: m.location.name == chapter)
+        if chapter_descriptor and not getattr(chapter_descriptor, 'is_entrance_exam', False) \
```

Comment:
These access checks could become part of the proposed `lms\djangoapps\courseware\entrance_exams.py` module